Python/ceval.c

```diff
@@ -4890,6 +4890,7 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, _PyInterpreterFrame *frame, int
 
         TARGET(CALL_PY_EXACT_ARGS) {
             assert(call_shape.kwnames == NULL);
+            DEOPT_IF(tstate->interp->eval_frame, CALL);
```

Comment:
I don't know how to verify that these opcodes (`CALL_PY_EXACT_ARGS` & `CALL_PY_WITH_DEFAULTS`) are the only ones where this check is needed - I trust you :)

Reply:
Specializations of `CALL` and `PRECALL` listed [here](https://github.com/python/cpython/blob/main/Lib/opcode.py#L256) and [here](https://github.com/python/cpython/blob/main/Lib/opcode.py#L300)